src/features/auth/services/auth-token-validation-service.ts

```diff
@@ -0,0 +1,36 @@
+import type { UserRepository } from '@/features/user/repositories/user-repository';
+import { UserNotFound } from '@/shared/errors/user-not-found-error';
+import type { JWTHelper, TokenPayload } from '@/shared/infra/jwt/jwt';
+import type { Service } from '@/shared/protocols/service';
+
+import type { AuthTokenValidationModel } from '../models/auth-token-validation-models';
+
+type Input = {
+  token: string;
+};
+
+type Output = {
+  email: string;
+  id: string;
+  name: null | string;
+  username: string;
+};
+
+export class AuthTokenValidationService implements Service<Input, Output> {
+  constructor(
+    private readonly userReposiotry: UserRepository,
+    private jwt: JWTHelper
+  ) {}
+
+  async execute({ token }: AuthTokenValidationModel) {
+    const payload = this.jwt.parseToken(token) as TokenPayload;
+
+    const isUserIdExist = await this.userReposiotry.findById(payload.userId);
+
+    if (!isUserIdExist) {
+      throw new UserNotFound();
+    }
+
```

Comment:
Adiciona uma etapa nesse service que seria a de atualizar esse usuario no banco de dados indicando que o e-mail foi validado. Cria uma coluna na tabela de usuário com o `confirmed` (é um exemplo) que nesse momento aqui você iria atualizar esse campo para true. 

Na criação desse campo, coloca como valor padrão =  `false`

Reply:
Não é melhor eu botar dateTime em vez de um boolean? Pq se o usuario trocar de novo, a gente vai ter um false. Na situação de data,  caso o usuario troque vamos ter o dia mais a confirmação que ele atualizou o email. 
Ele começa com Null e caso aconteça da atualização ele recebe um datatime, posso criar uma nova coluna com emailUpdated e a gente trabalha a logica nisso. O que acha?

Reply:
Nesse caso voce se refere a troca de email depois do usuario ja logado na plataforma. Essa etapa da atividade o usuario nao tem a conta criada ainda e sem a confirmaçao do email ele nao teria acesso a plataforma primeiramente. 

Eu lembrei hoje que já existe uma coluna na tabela `users` chamada `is_active`, que foi pensada nesse fluxo. 


Pensando nesse cenário que você levantou de alteração de email, daria para fazer outra abordagem, tipo: quando o usuário fizer a alteração do email na tela, a API recebe a requisição, salva isso em uma cache e envia o email de confirmação, quando for feita confirmação vai email, a API pega a informaçao do cache e efetiva a troca no banco de dados, trocando o email e atualizando o updated_at na tabela users.